matrix-tools/internal/pkg/args/args.go

```diff
@@ -110,26 +127,52 @@ func ParseArgs(args []string) (*Options, error) {
 		}
 		for _, generatedSecretArg := range strings.Split(*secrets, ",") {
 			parsedValue := strings.Split(generatedSecretArg, ":")
-			if len(parsedValue) != 3 {
-				return nil, fmt.Errorf("invalid generated secret format, expect <name:key:type>: %s", generatedSecretArg)
+			if len(parsedValue) < 3 {
+				return nil, fmt.Errorf("invalid generated secret format, expect <name:key:type:...>: %s", generatedSecretArg)
 			}
 			var parsedSecretType SecretType
 			if parsedSecretType, err = parseSecretType(parsedValue[2]); err != nil {
-				return nil, fmt.Errorf("invalid secret type in %s", generatedSecretArg)
+				return nil, fmt.Errorf("invalid secret type in %s : %v", generatedSecretArg, err)
 			}
 
 			generatedSecret := GeneratedSecret{ArgValue: generatedSecretArg, Name: parsedValue[0], Key: parsedValue[1], Type: parsedSecretType}
 			options.GeneratedSecrets = append(options.GeneratedSecrets, generatedSecret)
 		}
-		options.SecretLabels = make(map[string]string)
+		options.Labels = make(map[string]string)
 		if *secretsLabels != "" {
 			for _, label := range strings.Split(*secretsLabels, ",") {
 				parsedLabelValue := strings.Split(label, "=")
-				options.SecretLabels[parsedLabelValue[0]] = parsedLabelValue[1]
+				options.Labels[parsedLabelValue[0]] = parsedLabelValue[1]
 			}
 		}
-		options.SecretLabels["app.kubernetes.io/managed-by"] = "matrix-tools-init-secrets"
+		options.Labels["app.kubernetes.io/managed-by"] = "matrix-tools-init-secrets"
 		options.Command = GenerateSecrets
+	case "deployment-markers":
+		err := deploymentMarkersSet.Parse(args[2:])
+		if err != nil {
+			return nil, err
+		}
+		for _, deploymentMarkerArg := range strings.Split(*deploymentMarkers, ",") {
+			parsedValue := strings.Split(deploymentMarkerArg, ":")
+			if len(parsedValue) < 4 {
+				return nil, fmt.Errorf("invalid deployment marker format, expect <name:key:step:newValue:[allowedValues:..]>: %s", deploymentMarkerArg)
```

Comment:
```suggestion
				return nil, fmt.Errorf("invalid deployment marker format, expect <name:key:step:newValue:[allowedValues,..]>: %s", deploymentMarkerArg)
```

I think doing the `allowedValues` as a CSV would be better than a varargs